### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR adds instructions to `README.md` on how to set up the dev environment and run tests for `uarray`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,10 +61,39 @@ This project is in active development and not ready for production use. However,
...
+
+### Install
+
+`uarray` and all development dependencies can be installed via:
+
+```
+pip install -e ".[all]"
+```
+
+Note that uarray supports Python versions >= 3.5.
```
Comment (hameerabbasi) [inline README.md]: We need to tell people here that it currently requires `contextvars` to be installed manually for Python < 3.7.

File: README.md [hunk 2]
```diff
@@ -61,10 +61,39 @@ This project is in active development and not ready for production use. However,
...
+
+### Testing
+
+Tests can be run from the main uarray directory as follows:
+
+```
+pytest
+```
+
 ### Releases
```
Comment (hameerabbasi) [inline README.md]: This whole releases section is invalid now. 🙁

File: README.md (hunk 1 above)
Comment (hameerabbasi) [reply README.md]: I pushed to `master` so that this isn't required anymore.

**Comment by hameerabbasi** [comment]
I think this should all be in a `CONTRIBUTING.md` rather than the `README.md`.

File: README.md (hunk 2 above)
Comment (jrbourbeau) [reply README.md]: I'm not sure how releases are currently handled. I can remove this section altogether if you'd like.

Comment (hameerabbasi) [reply README.md]: That'd be nice, I use `twine`. See https://github.com/pydata/sparse/blob/master/release-procedure.md

**Pull request merged**